mapviz/include/mapviz/mapviz_plugin.h

```diff
@@ -318,7 +318,38 @@ public Q_SLOTS:
     draw_order_(0)
   {}
 
-  private:
+  void LoadQosConfig(const YAML::Node& node, rmw_qos_profile_t& qos, const std::string prefix = "") const
+  {
+    if (node[prefix + "qos_dpeth"])
+    {
+      qos.depth = node[prefix + "qos_depth"].as<int>();
+    }
+
+    if (node[prefix + "qos_history"])
+    {
+      qos.history = static_cast<rmw_qos_history_policy_e>(node[prefix + "qos_history"].as<int>());
+    }
+
+    if (node[prefix + "qos_reliability"])
+    {
+      qos.reliability = static_cast<rmw_qos_reliability_policy_e>(node[prefix + "qos_reliability"].as<int>());
+    }
+
+    if (node[prefix + "qos_durability"])
+    {
+      qos.durability = static_cast<rmw_qos_durability_policy_e>(node[prefix + "qos_durability"].as<int>());
+    }
+  }
+
+  void SaveQosConfig(YAML::Emitter& emitter, const rmw_qos_profile_t& qos, const std::string prefix = "") const
+  {
+    emitter << YAML::Key << prefix << "qos_depth" << YAML::Value << qos.depth;
```

Comment:
```suggestion
    emitter << YAML::Key << prefix + "qos_depth" << YAML::Value << qos.depth;
```
The `<<` operator was doing funky things here when I tried this out and concatenating the strings with just a `+` fixed it for me in local testing.